students/WesleyM/Lesson8/circle.py

```diff
@@ -0,0 +1,44 @@
+from io import StringIO
+import math
+
+class Circle:
+    def __init__(self, the_radius):
+        self.radius = the_radius
+        
+    @property
+    def diameter(self):
+        return self.radius*2
+
+    @diameter.setter
+    def diameter(self, diameter):
+        self.radius = diameter/2
+    
+    @property
+    def area(self):
+        return self.radius * 2 * math.pi
+    
+    @classmethod
+    def from_diameter(cls, diameter):
+        return cls(diameter/2)
+    
+    def __str__(self):
+        return "Circle with radius: {}".format(self.radius)
+    
+    def __repr__(self):
+        return "Circle({})".format(self.radius)
+    
+    def __add__(self, other):
+        return Circle(self.radius + other.radius)
+    
+    def __mul__(self, other):
+        return Circle(self.radius * other)
+    
+    def __lt__(self,other):
+        return self.radius < other.radius
+        
+    def __eq__(self, other):
+        return self.radius == other.radius
+        
```

Comment:
If you use the `@total_ordering` decorator from the `functools` module of the Python standard library to decorate your class, you only need to define `__eq__` and `__lt__`, otherwise, you should probably define the rest of the comparisons like `__ne__` for `!=`, `__le__` for `<=` etc.